src/OrleansRuntime/Scheduler/WorkItemGroup.cs

```diff
@@ -453,3 +467,76 @@ private void ReportWorkGroupProblem(string what, ErrorCode errorCode)
         }
     }
 }
+
+// todo: for now in this file in order to avoid merging
+// just simple descendant of task, used for distinguishing of work item group execute method from it's work items
+public class WorkItemGroupExecuteTask : Task
+{
+    public WorkItemGroupExecuteTask(Action action) : base(action)
+    {
+    }
+   
+    public WorkItemGroupExecuteTask(Action action, CancellationToken cancellationToken) : base(action, cancellationToken)
+    {
+    }
+
+    public WorkItemGroupExecuteTask(Action action, TaskCreationOptions creationOptions) : base(action, creationOptions)
+    {
+    }
+
+    public WorkItemGroupExecuteTask(Action action, CancellationToken cancellationToken, TaskCreationOptions creationOptions) : base(action, cancellationToken, creationOptions)
+    {
+    }
+
+    public WorkItemGroupExecuteTask(Action<object> action, object state) : base(action, state)
+    {
+    }
+
+    public WorkItemGroupExecuteTask(Action<object> action, object state, CancellationToken cancellationToken) : base(action, state, cancellationToken)
+    {
+    }
+
+    public WorkItemGroupExecuteTask(Action<object> action, object state, TaskCreationOptions creationOptions) : base(action, state, creationOptions)
+    {
+    }
+
+    public WorkItemGroupExecuteTask(Action<object> action, object state, CancellationToken cancellationToken, TaskCreationOptions creationOptions) : base(action, state, cancellationToken, creationOptions)
+    {
+    }
+}
+public static class TaskGetter
+{
+    private static string _propertyName;
+    private static Type _taskType;
+    private static PropertyInfo _property;
+    private static Func<Task> _getter;
+
+    static TaskGetter()
+    {
+        _taskType = typeof(Task);
+        _propertyName = "InternalCurrent";
+        SetupGetter();
+    }
+
+    public static void SetPropertyName(string newName)
+    {
+        _propertyName = newName;
+        SetupGetter();
+    }
+
+    public static Task CurrentTask => _getter();
```

Comment:
This is not really ready for review yet. This particular code was used for ensuring that current task and task scheduler is corrent during work items executions, and was not removed as I thought that this checks possibly could be hidden behind conditional compilation variable.